### PR TITLE
Add thnn_conv_transpose2d_backward to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/convolution.cpp [hunk 1]
```diff
@@ -234,6 +234,16 @@ xla::XlaOp BuildThnnConv2dBackwardWeight(
...
 }
 
+xla::XlaOp BuildGradBias(xla::XlaOp grad_output) {
+  xla::Shape grad_output_shape = XlaHelpers::ShapeOfXlaOp(grad_output);
+  return xla::Reduce(
+      grad_output,
+      XlaHelpers::ScalarValue<float>(0, grad_output_shape.element_type(),
+                                     grad_output.builder()),
+      XlaHelpers::CreateAddComputation(grad_output_shape.element_type()),
+      {0, 2, 3});
```
Comment (dlibenzi) [inline torch_xla/csrc/convolution.cpp]: Would be good to have some comment around these magic reduce dimensions. Same for the Transpose 10 LOC above.

File: torch_xla/csrc/ops/conv_transpose2d_backward.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+namespace {
+
+xla::Shape NodeOutputShape(
+    const Value& grad_output, const Value& input, const Value& weight,
+    tensorflow::gtl::ArraySlice<const xla::int64> stride,
+    tensorflow::gtl::ArraySlice<const xla::int64> padding) {
+  auto lower_for_shape_fn =
+      [stride, padding](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK_EQ(operands.size(), 3)
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/conv_transpose2d_backward.cpp]: XLA_CHECK_EQ(operands.size(), 3);

File: torch_xla/csrc/ops/conv_transpose2d_backward.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  XlaOpVector Lower(LoweringContext* loctx) const override;
+
+  std::string ToString() const override;
+
+  const std::vector<xla::int64>& stride() const { return stride_; }
+
+  const std::vector<xla::int64>& padding() const { return padding_; }
+
+ private:
+  // The parameters of the convolution.
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/conv_transpose2d_backward.h]: comment--

File: torch_xla/csrc/tensor_methods.cpp [hunk 4]
```diff
@@ -622,6 +623,22 @@ XLATensor XLATensor::conv_transpose2d(const XLATensor& input,
...
 
+std::tuple<XLATensor, XLATensor, XLATensor>
+XLATensor::conv_transpose2d_backward(const XLATensor& out_backprop,
+                                     const XLATensor& input,
+                                     const XLATensor& weight,
+                                     std::vector<xla::int64> stride,
+                                     std::vector<xla::int64> padding) {
+  ir::NodePtr node = ir::MakeNode<ir::ops::ConvTranspose2dBackward>(
+      out_backprop.GetIrValue(), input.GetIrValue(), weight.GetIrValue(),
+      stride, padding);
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: std::move() stride&padding

File: torch_xla/csrc/convolution.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/convolution.cpp]: Done.

Comment (asuhan) [reply torch_xla/csrc/convolution.cpp]: Done.

File: torch_xla/csrc/ops/conv_transpose2d_backward.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/conv_transpose2d_backward.cpp]: Done.

File: torch_xla/csrc/ops/conv_transpose2d_backward.h (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/ops/conv_transpose2d_backward.h]: Done.

File: torch_xla/csrc/tensor_methods.cpp (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Done.

**Comment by asuhan** [comment]
FYI, had to scale back the test size quite a bit to keep the execution time manageable.

**Pull request merged**